tests/array_test.cpp

```diff
@@ -0,0 +1,122 @@
+#include <catch2/catch.hpp>
+
+#include "ara/core/array.h"
+
+TEST_CASE("Array can be constructed", "[SWS_CORE], [SWS_CORE_01201]")
+{
+    ara::core::Array<int, 1> array{0};
+
+    REQUIRE(array[0] == 0);
+}
+
+TEST_CASE("Array.at()", "[SWS_CORE], [SWS_CORE_01201]")
+{
+    ara::core::Array<int, 1> array{0};
+    REQUIRE(array.at(0) == 0);
```

Comment:
I'd also add a check for exception case i.e.:
```suggestion
    REQUIRE(array.at(0) == 0);
    CHECK_THROWS(array.at(1), std::out_of_range);
```